lib/internal/modules/esm/utils.js

```diff
@@ -165,4 +173,6 @@ module.exports = {
   initializeHooks,
   getDefaultConditions,
   getConditionsSet,
+  loaderWorkerId: 'internal/modules/esm/worker',
```

Comment:
It's definitely not a path (because the internal module loader doesn't use paths), I think calling it an ID is more correct.

Reply:
O.o it is literally the path to the file (but with CJS's whacked exclusion of the file extension) and is even consumed as such: https://github.com/nodejs/node/blob/3707d0051fbba829cadb2fccac2a7929eae04bad/lib/internal/modules/esm/hooks.js#L498

Reply:
It's not the path to a file, because internal modules are not file, it's more correct to think of them as an offset in the `node` binary. `InternalWorker` constructor doesn't take a path, it only accepts internal module IDs – you can try to replace `loaderWorkerId` with a path at the line you linked to to convince yourself if you want.

Reply:
Yes, it’s whatever we call the string that’s the input to `internalBinding()`. The signature of that function is `internalBinding(module)`, so I guess we could use `loaderWorkerModule`. Not sure if that’s any better than `loaderWorkerId`. It’s not really a specifier because we don’t use this with `import`.